test/lit.cfg

```diff
@@ -311,6 +311,40 @@ if test_sdk_overlay_dir is not None:
 test_options = os.environ.get('SWIFT_TEST_OPTIONS')
 config.swift_test_options = ' ' + test_options if test_options else ''
 
+def get_clang_library_dir():
+    llvm_lib_clang = os.path.join(config.llvm_libs_dir, 'clang')
+    if not os.path.exists(llvm_lib_clang):
+        return ''
+
+    versions = os.listdir(llvm_lib_clang)
+    if not len(versions):
+        return ''
+
+    version = max(versions)
+    return os.path.join(llvm_lib_clang, version, 'lib',
+                        platform.system().lower())
+
+config.clang_library_dir = get_clang_library_dir()
```

Comment:
I hate to say it, but this is the wrong path: we should be looking in `lib/swift/clang/` rather than `lib/clang/$VERSION/`. More than that, though, please test with Xcode as well; Xcode builds put the "lib" directory in a subdirectory named after the CMake configuration (Debug, Release, RelWithDebInfo, MinSizeRel), and we should be able to handle that. (You shouldn't need to do anything special; it should just be the existing Swift resource dir.)